miden-lib/asm/sat/account.masm

```diff
@@ -174,6 +177,27 @@ export.is_immutable_account
     # => [is_immutable_account]
 end
 
+#! Validates an account id. Panics if the account id is invalid.
+#! Account id must have at least `MIN_ACCOUNT_ONES` ones.
+#!
+#! Stack: [acct_id]
+#! Output: []
+#!
+#! - acct_id is the account id. 
+export.validate_id
+    # split felt into 32 bit limbs
+    u32split
+    # => [l_1, l_0]
+
+    # count the number of 1 bits
+    u32unchecked_popcnt swap u32unchecked_popcnt add
+    # => [ones]
+
+    # check if the number of ones is at least MIN_ACCOUNT_ONES ones.
+    push.MIN_ACCOUNT_ONES gte assert
```

Comment:
I would use `u32unchecked_gte` here as it is quite a bit more efficient (and I think we are guaranteed to have number of ones be within u32 range).